### DISCUSSION TITLE
autodiscoverFilter - missing subgroups

### DISCUSSION THREAD

**Discussion by adam-moss** [renovatebot/renovate#10733] [Request Help]
Hi,

Query about autodiscoverfilter.  I'm using this on gitlab.com so have it set to the pattern `/my-group/!(my-attic)**/*`.

If my understanding is correct it should renovate everything in `/my-group` with the exception of `/my-group/my-attic`.

However the behaviour I'm observing is nest subgroups (i.e. /my-group/group-1/group-2/group-3/project` don't appear to be getting renovated - there are no MR's created nor is the dependency dashboard.

Are there any known issues here, or have I misunderstood how the filter works?

Thanks!

**Comment by viceice** [comment]
Please check the `DEBUG`  logs, renovate should print all discovered repos.

Maybe renovate has missing permissions on some repos.

https://github.com/renovatebot/renovate/blob/d0434d5df1516fdb2ef29917df41564153f5330c/lib/workers/global/autodiscover.ts#L23-L43

This is the query for repos. access level: `Developer=30`

https://github.com/renovatebot/renovate/blob/4a99883048843fbb7504bd7cd47fb7b73540156c/lib/platform/gitlab/index.ts#L127-L127

Also mirror and archived repos are skiped:
https://github.com/renovatebot/renovate/blob/4a99883048843fbb7504bd7cd47fb7b73540156c/lib/platform/gitlab/index.ts#L133-L133

**Reply by adam-moss** [reply]
Awesome, thanks let me look into the log.

Its definitely not permissions as I'm running it with an `Owner` level token on the parent group, so that's inherited through the entire tree.

On that mirroring and archived skipping, very useful info, thank you :)

**Reply by viceice** [reply]
@rarkins @HonkingGoose Maybe we should add those filter info to the platform specific docs?

**Reply by HonkingGoose** [reply]
If we think this information is helpful for more people and will be asked about more often, we should put it in the docs. 😉 

The current [docs on `autodiscoverFilter`](https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter) only mentions that it takes a minimatch glob-style pattern, but nothing else.

> <html><body>
> <h2 id="autodiscoverfilter">autodiscoverFilter</h2>
> <p>Filter the list of autodiscovered repositories.</p>
> <div class="md-typeset__scrollwrap"><div class="md-typeset__table">
> 
> Name | Value
> -- | --
> type | string
> 
> </div></div>
> <p>You can use this option to filter the list of repositories that the Renovate bot account can access through <code>autodiscover</code>.
> It takes a <a href="https://www.npmjs.com/package/minimatch">minimatch</a> glob-style pattern.</p>
> <p>e.g.</p>
> <pre id="__code_4"><code class="json">{
>   "autodiscoverFilter": "project/*"
> }
> </code></pre>
> </body>
> </html>

**Reply by viceice** [reply]
yes, i don't want to change autodiscover, as it works as intended.

But platforms do some filtering before that, so i would like to have documented at https://docs.renovatebot.com/modules/platform/

**Reply by viceice** [reply]
Maybe we should include the `lib/platform/xxx/index.md` files too if available?

**Reply by adam-moss** [reply]
Ok, looking at the info log I can see I'm only seeing repos `/my-group/1st-subgroup/project`.  I'm not seeing anything for nested subgroups.

As a slight aside, you can possibly optimise that api call at line 127 to include the `archived` filter and remove that from the `filter` at 133.

**Reply by viceice** [reply]
@adam-moss Feel free to open a pr to change this 🙃

**Reply by HonkingGoose** [reply]
@viceice I don't know enough about this to add to the docs, but feel free to open a PR and ping me for a review. 😉
